crates/aquascope/src/mir/utils.rs

```diff
@@ -113,6 +124,30 @@ impl<'tcx> TyExt<'tcx> for Ty<'tcx> {
       _ => None,
     })
   }
+
+  fn does_implement_trait(
+    &self,
+    tcx: TyCtxt<'tcx>,
+    param_env: ParamEnv<'tcx>,
+    trait_def_id: DefId,
+  ) -> bool {
+    use rustc_infer::traits::EvaluationResult;
+
+    let infcx = tcx.infer_ctxt().build();
+    let ty = tcx.erase_regions(*self);
+    let result = infcx.type_implements_trait(trait_def_id, [ty], param_env);
+    matches!(
+      result,
+      EvaluationResult::EvaluatedToOk
+        | EvaluationResult::EvaluatedToOkModuloRegions
+    )
+  }
+
+  fn is_copyable(&self, tcx: TyCtxt<'tcx>, param_env: ParamEnv<'tcx>) -> bool {
```

Comment:
We should probably use [`Ty::is_copy_modulo_regions`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.Ty.html#method.is_copy_modulo_regions) instead.